### PR TITLE
[sc-3756] Add inline code custom plugin

### PR DESCRIPTION
**Pull request by Bondok6**
The current situation:

Users have to click on Insert HTML or switch to the Source Mode and create something like this:

```html
<p> Some text <code> inline code </code> more text </p>
```

[The Card](https://app.shortcut.com/helpjuice/story/3756/ck-improvement-inline-code)

In this PR:

I added a new custom plugin called `inline code`. This plugin provides a more user-friendly approach for adding inline code within the content. 
Users can now select the desired text and click on the `Insert Inline Code` icon to add inline code to their content

[Screencast from 13 يول, 2023 EEST 10:54:00 ص.webm](https://github.com/Helpjuice/ckeditor5/assets/45231276/116ee49b-ba5c-47a4-a39e-c007312c44ea)

### CODE REVIEW

**Comment by Bondok6** [comment]
Update: I changed the position of the `insert inline code` button to be next to the `insert code block` button

image

File: packages/ckeditor5-build-helpjuice-v2/src/plugins/inlinecode/inlinecodecommand.ts
Comment (NicholasJarr) [inline packages/ckeditor5-build-helpjuice-v2/src/plugins/inlinecode/inlinecodecommand.ts:1]: `AttributeCommand` already exists in CKEditor. Is there any difference between this implementation and the CKEditor one?

If not, I think we can simplify this by removing this command and registering an instance of `AttributeCommand` instead:

```
editor.commands.add( INLINECODE, new AttributeCommand( editor, INLINECODE ) );
```

File: packages/ckeditor5-build-helpjuice-v2/src/plugins/inlinecode/inlinecodeediting.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+		editor.model.schema.extend( '$text', { allowAttributes: INLINECODE } );
+
+		editor.conversion.attributeToElement( {
+			model: INLINECODE,
+			view: {
+				name: 'code'
+			}
+		} );
+
+		editor.commands.add( INLINECODE, new InlineCodeCommand( editor, INLINECODE ) );
```
Comment (NicholasJarr) [inline packages/ckeditor5-build-helpjuice-v2/src/plugins/inlinecode/inlinecodeediting.ts]: Minor, but I would use a different constant than the model name.

**Review by NicholasJarr — COMMENTED** [review]
I tested it and it worked without issues 👍🏻 . Just left some suggestions.

BTW, is there a different icon for this? Maybe something with a code icon?

File: packages/ckeditor5-build-helpjuice-v2/src/plugins/inlinecode/inlinecodecommand.ts
Comment (Bondok6) [reply packages/ckeditor5-build-helpjuice-v2/src/plugins/inlinecode/inlinecodecommand.ts:1]: There's no difference.
I removed this file and imported `AttributeCommand` from the CKEditor

File: packages/ckeditor5-build-helpjuice-v2/src/plugins/inlinecode/inlinecodeediting.ts (hunk 1 above)
Comment (Bondok6) [reply packages/ckeditor5-build-helpjuice-v2/src/plugins/inlinecode/inlinecodeediting.ts]: Done

**Comment by Bondok6** [comment]
> I tested it and it worked without issues 👍🏻 . Just left some suggestions.
> 
> BTW, is there a different icon for this? Maybe something with a code icon?

Thanks @NicholasJarr for your suggestions
 
- I changed the icon and added a shortcut "Ctrl + Alt + C"


[Screencast from 17 يول, 2023 EEST 09:19:22 ص.webm](https://github.com/Helpjuice/ckeditor5/assets/45231276/661cf1b4-4942-42f2-9793-503cfa5f0d97)

File: packages/ckeditor5-build-helpjuice-v2/src/plugins/inlinecode/inlinecodeui.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+ * @license Copyright (c) 2003-2023, CKSource Holding sp. z o.o. All rights reserved.
+ * For licensing, see LICENSE.md or https://ckeditor.com/legal/ckeditor-oss-license
+ */
+
+import { Plugin } from 'ckeditor5/src/core';
+import { ButtonView } from 'ckeditor5/src/ui';
+import type AttributeCommand from '@ckeditor/ckeditor5-basic-styles/src/attributecommand';
+
+const INLINECODE = 'inlineCode';
+const INLINECODE_COMMAND = 'inlineCodeCommand';
```
Comment (NicholasJarr) [inline packages/ckeditor5-build-helpjuice-v2/src/plugins/inlinecode/inlinecodeui.ts]: Minor, but can we reuse the definitions from `inlinecodeediting.ts`. This way, we need to change only one place in order to update the constant value.

**Review by NicholasJarr — COMMENTED** [review]
@Bondok6

I tried it out and I noticed a small issue. If I add a inline code, change to the source editor, and then I switch back to the editor view, I can't toggle the inline code anymore:

https://github.com/Helpjuice/ckeditor5/assets/28270050/78e2f288-eca7-471f-a8f9-cdfbb6bfdd60

This is the code I'm using:

```html
<p>
    testing this stuff: <code>console.log('test');</code> .
</p>
```

**Comment by Bondok6** [comment]
> @Bondok6
> 
> I tried it out and I noticed a small issue. If I add a inline code, change to the source editor, and then I switch back to the editor view, I can't toggle the inline code anymore:
> 
>  2023-07-17.12-45-00.mp4 
> This is the code I'm using:
> 
> ```
> <p>
>     testing this stuff: <code>console.log('test');</code> .
> </p>
> ```

Hey @NicholasJarr , you have to **select the text** again and then click on the icon to toggle it 
That's how it works. ( in the same way for "Bold, Italic, ..." plugins )

> The text should be selected

[Screencast from 17 يول, 2023 EEST 11:32:00 م.webm](https://github.com/Helpjuice/ckeditor5/assets/45231276/027bc22f-5ca3-4048-b1e0-3539621f34cc)

File: packages/ckeditor5-build-helpjuice-v2/src/plugins/inlinecode/inlinecodeui.ts (hunk 2 above)
Comment (Bondok6) [reply packages/ckeditor5-build-helpjuice-v2/src/plugins/inlinecode/inlinecodeui.ts]: Done

**Review by NicholasJarr — COMMENTED** [review]
> Hey @NicholasJarr , you have to **select the text** again and then click on the icon to toggle it That's how it works. ( in the same way for "Bold, Italic, ..." plugins )

My bad. I thought it worked differently.

I have just one last suggestion. We can't remove it through the Remove Format feature (like we can with bold, italics, etc.). Can you change Remove Format to allow it?

If I'm not mistaken, you only need to add this code after the call for `schema.extend` in `inlinecodeediting.ts`:

```js
 editor.model.schema.setAttributeProperties( INLINECODE, {
        isFormatting: true
    } );
```

Can you check if that works?

**Comment by Bondok6** [comment]
@NicholasJarr That's a good point! 
Let me try to add it.

**Comment by Bondok6** [comment]
@NicholasJarr It works well, thanks

[Screencast from 18 يول, 2023 EEST 05:34:08 م.webm](https://github.com/Helpjuice/ckeditor5/assets/45231276/65781d32-f4eb-4621-a428-9d4780166291)

**Review by NicholasJarr — APPROVED** [review]

**Pull request merged**